components/sup/src/main.rs

```diff
@@ -60,16 +60,20 @@ static RING_ENVVAR: &'static str = "HAB_RING";
 static RING_KEY_ENVVAR: &'static str = "HAB_RING_KEY";
 
 fn main() {
-    env_logger::init().unwrap();
-    enable_features_from_env();
+    boot();
     if let Err(e) = start() {
         println!("{}", e);
         std::process::exit(1);
     }
 }
 
+fn boot() {
+    env_logger::init().unwrap();
+    enable_features_from_env();
+    crypto::init();
```

Comment:
Looks like we were ignoring the return value from sodiumoxide in our helper function - I've refactored this and put a commit on top of the rest with the change